### PR TITLE
Console application

### PR DESCRIPTION
**Pull request by ihhyy**

### CODE REVIEW

File: BL/Services/WeatherServices.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    public class WeatherServices : IWeatherService
+    {
+        static readonly HttpClient client = new HttpClient();
+        private readonly string APIKey = "2a1e352deac76c1b856f00075b07cd74";
+
+        public async Task<string> GetWeatherByCytyName(string cityName)
+        {
+            HttpResponseMessage response = await client.GetAsync("https://api.openweathermap.org/data/2.5/weather?q=" + cityName + "&appid=" + APIKey + "&units=metric");
+            var responceBody = await response.Content.ReadAsStringAsync();
+
```
Comment (alesyalis) [inline BL/Services/WeatherServices.cs:18]: The above logic is DAL, below is BL

File: BL/Services/WeatherServices.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+namespace BL.Services
+{
+    public class WeatherServices : IWeatherService
+    {
+        static readonly HttpClient client = new HttpClient();
+        private readonly string APIKey = "2a1e352deac76c1b856f00075b07cd74";
+
+        public async Task<string> GetWeatherByCytyName(string cityName)
+        {
+            HttpResponseMessage response = await client.GetAsync("https://api.openweathermap.org/data/2.5/weather?q=" + cityName + "&appid=" + APIKey + "&units=metric");
```
Comment (alesyalis) [inline BL/Services/WeatherServices.cs]: Move  key to the config file

File: BL/Interfaces/IWeatherService.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿using System.Threading.Tasks;
+
+namespace BL.Interfaces
+{
+    public interface IWeatherService
+    {
+        Task<string> GetWeatherByCytyName(string cityName);
```
Comment (denja2) [inline BL/Interfaces/IWeatherService.cs]: Async

File: DAL/Repositories/WeatherRepository.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DAL.Repositories
+{
+    public class WeatherRepository : IWeatherRepository
+    {
+
+        private readonly string APIKey;
+        private readonly HttpClient client;
```
Comment (denja2) [inline DAL/Repositories/WeatherRepository.cs]: Naming convention

File: DAL/Repositories/WeatherRepository.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+        public WeatherRepository()
+        {
+            APIKey = ConfigurationManager.AppSettings["APIKey"];
+            client = new HttpClient();
```
Comment (denja2) [inline DAL/Repositories/WeatherRepository.cs]: Inject

File: DAL/Repositories/WeatherRepository.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        }
+
+        public async Task<Weather> GetWeatherByCityNameAsync(string cityName)
+        {
+            HttpResponseMessage response = await client.GetAsync("https://api.openweathermap.org/data/2.5/weather?q=" + cityName + "&appid=" + APIKey + "&units=metric");
```
Comment (denja2) [inline DAL/Repositories/WeatherRepository.cs]: Var

File: DAL/Repositories/WeatherRepository.cs [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+            var responceBody = await response.Content.ReadAsStringAsync();
```
Comment (denja2) [inline DAL/Repositories/WeatherRepository.cs]: Read response status

File: BL/Services/WeatherServices.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        private IWeatherRepository _weatherRepositiry;
+
+        public WeatherServices(IWeatherRepository weatherRepository)
+        {
+            _weatherRepositiry = weatherRepository;
+        }
+
+        public async Task<string> GetWeatherByCytyName(string cityName)
+        {
```
Comment (denja2) [inline BL/Services/WeatherServices.cs:20]: Validator

**Review by denja2 — CHANGES_REQUESTED** [review]

File: BL/Interfaces/IWeatherService.cs (hunk 3 above)
Comment (ihhyy) [reply BL/Interfaces/IWeatherService.cs]: Done

File: DAL/Repositories/WeatherRepository.cs (hunk 6 above)
Comment (ihhyy) [reply DAL/Repositories/WeatherRepository.cs]: Done

File: DAL/Repositories/WeatherRepository.cs (hunk 4 above)
Comment (ihhyy) [reply DAL/Repositories/WeatherRepository.cs]: Done

File: DAL/Repositories/WeatherRepository.cs (hunk 5 above)
Comment (ihhyy) [reply DAL/Repositories/WeatherRepository.cs]: Done

**Review by Xodoc — APPROVED** [review]

**Review by alesyalis — APPROVED** [review]

**Pull request merged**